java/flight/flight-core/src/test/java/org/apache/arrow/flight/FlightTestUtil.java

```diff
@@ -51,7 +52,10 @@ public static <T> T getStartedServer(Function<Location, T> newServerFromLocation
     IOException lastThrown = null;
     T server = null;
     for (int x = 0; x < 3; x++) {
```

Comment:
Kept the retry in case the port gets before the lambda is applied and the server is created.